plugins/anghami/integration.js

```diff
@@ -0,0 +1,66 @@
+var playbackStatus = mellowplayer.PlaybackStatus.STOPPED;
+
+function update() {
+    return {
+        "playbackStatus": playbackStatus,
+        "canSeek": false,
+        "canGoNext": true,
+        "canGoPrevious": true,
+        "canAddToFavorites": false,
+        "volume": 1,
+        "duration": 0,
```

Comment:
Are you sure you cannot get song duration and position? It is easy to convert a duration string to a number of seconds using the below function:

```javascript
function toSeconds(string) {
    try {
        var dtimes = string.split(":");
        var dminutes = dtimes[0];
        var dseconds = dtimes[1];
        var duration = parseInt(dseconds, 10) + (parseInt(dminutes, 10) * 60);
    } catch (e) {
        var duration = 0;
    }

    return duration
}
```

Reply:
Anghami prints elapsed and remaining time only, so I think I can get the position easily, but the duration is going to need a bit of work.

Player shown below

![image](https://user-images.githubusercontent.com/619795/30657908-579200f0-9e39-11e7-8002-b288bea8464b.png)

Reply:
I did a similar thing for Mixcloud. I just added the elapsed time to the remaining time to get the duration.

Reply:
@ColinDuquesnoy Okay, I'll do that then.